enterprise_subsidy/apps/fulfillment/api.py

```diff
@@ -1,34 +1,117 @@
+        pass
+
+
+class GEAGFulfillmentHandler(ExternalFulfillmentHandler):
+    CENTS_PER_DOLLAR = 100.0
+
+    def get_smarter_client(self):
+        return GetSmarterEnterpriseApiClient(
+            client_id=settings.GET_SMARTER_OAUTH2_KEY,
+            client_secret=settings.GET_SMARTER_OAUTH2_SECRET,
+            provider_url=settings.GET_SMARTER_OAUTH2_PROVIDER_URL,
+            api_url=settings.GET_SMARTER_API_URL
+        )
+    
+    def _get_geag_transaction_price(self, transaction):
+        return float(transaction.quantity) / self.CENTS_PER_DOLLAR
+    
+    def _get_geag_variant_id(self):
+        self.content_summary.get('variant_id')
 
-    metadata = content_metadata_api.get_content_metadata(content_key)
-    course_type = metadata.get('course_type')
-    if course_type in EXEC_ED_2U_COURSE_TYPES:
-        # really we need to return an exec-ed-capable client
-        return None
-    if course_type in OPEN_COURSES_COURSE_TYPES:
-        # return an edx-enterprise client
-        return None
-    return None
+    def _create_allocation_payload(self, transaction, currency='USD'):
+        return {
+            'payment_reference': transaction.uuid,
+            'enterprise_customer_uuid': self.subsidy.enterprise_customer_uuid,
+            'currency': currency,
+            'order_items': [
+                {
+                    # productId will be the variant id from product details
+                    'productId': self._get_geag_variant_id(),
+                    'quantity': 1,
+                    # TODO prices are in dollars or cents?
+                    'normalPrice': self._get_geag_transaction_price(transaction),
+                    'discount': 0.0,
+                    'finalPrice': self._get_geag_transaction_price(transaction)
+                }
+            ],
+            # TODO we need this passed in
+            'first_name': '',
+            'last_name': '',
+            'date_of_birth': "2021-05-12",
+            'terms_accepted_at': "2021-05-21T17:32:28Z"
+        }
+
+    def fulfill(self, transaction):
+        # call GEAG client
+        # init GEAG client
+        # allocation_id = geag_client.allocate(...)
+        # enterprise_client.enroll(...)
+        # return allocation_id, edx_fulfillment_uuid, maybe?
+        # but then also do an enroll in edx-enterprise
+        # maybe via a sub-OpenCourseFulfillmentHandler?
+        pass
+
+
+class FulfillmentManager:
+    """
+    Manager for creating an appropriate handler
+    and storing any log-level facts about the fulfillment attempt.
+    """
+
+
+def get_fulfillment_handlers(content_metadata):
+    """
+    Could implement this as multiple fulfillment operations, but all on a single
+    transaction record to fulfill.
     """
-    raise NotImplementedError
+    if content_metadata.get('product_source') == 'the-twou-product-source':
+        return [GEAGFulfillmentHandler(content_metadata), OpenCourseFulfillmentHandler(content_metadata)]
```

Comment:
the `OpenCourseFulfillmentHandler` is different than other fulfillment providers because our "commit transaction" relies on the enterprise enrollment. i think we should just call this explicitly every time and not include it in a manager like this